autogen/main/variables.tf.tmpl

```diff
@@ -766,5 +766,23 @@ variable "enable_identity_service" {
   description = "Enable the Identity Service component, which allows customers to use external identity providers with the K8S API."
   default     = false
 }
+
+variable "protect_config" {
```

Comment:
Suggest to simplify the variable for this configuration. Do you reckon we could `workload_config_audit_mode` and 
`workload_vulnerability_mode` variables to toggle the feature?

Reply:
My main reasoning behind use an object is that I don't know if GCP will add any other API that we might want to add in the future, and then I thought it would be easier to do that change with an object without making any breaking change.
But a string is definitely easier. I have updated the PR to match your suggestion.